app/demo-loop/LDemoIO.root.cc

```diff
@@ -7,15 +7,64 @@
 //---------------------------------------------------------------------------//
 #include "LDemoIO.hh"
 
+#include <string>
+#include <vector>
 #include <TBranch.h>
 #include <TFile.h>
 #include <TTree.h>
 
+#include "celeritas/global/ActionRegistry.hh"
+
 namespace demo_loop
 {
 //---------------------------------------------------------------------------//
+/*!
+ * Store CoreParams data to the ROOT MC truth output file.
+ * Called by `to_root(...)`.
+ *
+ * \note
+ * Currently only storing the action labels so their IDs can be
+ * identified. If other parameters are needed for future
+ * debugging/analyses, this function can easily be expanded.
+ */
+void store_core_params(std::shared_ptr<celeritas::RootFileManager>& root_manager,
+                       celeritas::CoreParams core_params)
+{
+    auto action_reg = core_params.action_reg();
+    CELER_EXPECT(action_reg);
```

Comment:
I'm not a huge fan of capturing shared pointers with `auto` because it's hard to tell how to use it. Also since core params are guaranteed to return non-null shared pointers, just do `auto const& action_reg = *core_params.action_reg();`. Then the reader can tell it's a regular object rather than a pointer-like one. (Drop the EXPECT and you should be using an ASSERT here anyway.)